testdata/testinput5

```diff
@@ -2337,6 +2337,217 @@
 
 # End caseless restrict tests
 
+# TESTS for PCRE2_EXTRA_TURKISH_CASING - again, tests with and without.
+
+/i/i,utf
+    i
+    I
+\= Expect no match
+    \x{0130}
+    \x{0131}
+
+/i/i,utf,turkish_casing
+    i
+    \x{0130}
+\= Expect no match
+    I
+    \x{0131}
+
+/I/i,utf
+    i
+    I
+\= Expect no match
+    \x{0130}
+    \x{0131}
+
+/I/i,utf,turkish_casing
+    I
+    \x{0131}
+\= Expect no match
+    i
+    \x{0130}
+
+/\x{0130}/i,utf
+    \x{0130}
+\= Expect no match
+    i
+    I
+    \x{0131}
+
+/\x{0130}/i,utf,turkish_casing
+    i
+    \x{0130}
+\= Expect no match
+    I
+    \x{0131}
+
+/\x{0131}/i,utf
+    \x{0131}
+\= Expect no match
+    i
+    I
+    \x{0130}
+
+/\x{0131}/i,utf,turkish_casing
+    I
+    \x{0131}
+\= Expect no match
+    i
+    \x{0130}
+
+/[i]/i,utf
+    i
+    I
+\= Expect no match
+    \x{0130}
+    \x{0131}
+
+/[i]/i,utf,turkish_casing
+    i
+    \x{0130}
+\= Expect no match
+    I
+    \x{0131}
+
+/[\x{0130}]/i,utf
+    \x{0130}
+\= Expect no match
+    i
+    I
+    \x{0131}
+
+/[\x{0130}]/i,utf,turkish_casing
+    i
+    \x{0130}
+\= Expect no match
+    I
+    \x{0131}
+
+/[\x{0120}-\x{0130}]/i,utf
+    \x{0130}
+\= Expect no match
+    i
+    I
+    \x{0131}
+
+/[\x{0120}-\x{0130}]/i,utf,turkish_casing
+    i
+    \x{0130}
+\= Expect no match
+    I
+    \x{0131}
+
+/[zi]/i,utf
+    i
+    I
+\= Expect no match
+    \x{0130}
+    \x{0131}
+
+/[zi]/i,utf,turkish_casing
+    i
+    \x{0130}
+\= Expect no match
+    I
+    \x{0131}
+
+/[z\x{0130}]/i,utf
+    \x{0130}
+\= Expect no match
+    i
+    I
+    \x{0131}
+
+/[z\x{0130}]/i,utf,turkish_casing
+    i
+    \x{0130}
+\= Expect no match
+    I
+    \x{0131}
+
+/[iI]/i,utf
+    i
+    I
+\= Expect no match
+    \x{0130}
+    \x{0131}
+
+/[iI]/i,utf,turkish_casing
+    i
+    I
+    \x{0130}
+    \x{0131}
+
+/[i\x{0130}]/i,utf
+    i
+    I
+    \x{0130}
+\= Expect no match
+    \x{0131}
+
+/[i\x{0130}]/i,utf,turkish_casing
+    i
+    \x{0130}
+\= Expect no match
+    I
+    \x{0131}
+
+/(.) \1/i,utf
+    i I
+\= Expect no match
+    i \x{0130}
+    \x{0131} I
+
+/(*TURKISH_CASING)(.) \1/i,utf
+    i \x{0130}
+    \x{0131} I
+\= Expect no match
+    i I
+
+/(.) \1/i,utf,turkish_casing
+    i \x{0130}
+    \x{0131} I
+\= Expect no match
+    i I
+
+/(.) \1/i,ucp
+    i I
+
+/(.) \1/i,ucp,turkish_casing
+\= Expect no match
+    i I
+
+/[a-z][^i]I/utf,caseless_restrict,turkish_casing
```

Comment:
Do we have caseless_restrict,turkish_casing with backreferences?

Reply:
There seem to be some blind spots there, added more tests in fad3d34b118bd91b214dd001e49c110ad17fd0bb that sadly also show that turkish_casing still has the bug that reported earlier.

```
PCRE2 version 10.45-DEV 2024-06-09 (8-bit)
  re> /[^i]/Bi,utf,turkish_casing
------------------------------------------------------------------
        Bra
        not clist 0049 0131
        Ket
        End
------------------------------------------------------------------
data> i
 0: i
```

Reply:
!!! Thank you Carlo, I've added your test - this case was covered by an existing test, but I'd got the test results flipped!

There was a bug. I had `caseset = UCD_DOTTED_I(meta)? ...` instead of `caseset = UCD_DOTTED_I(c)? ...` due to a copy-paste bug.

Fixed, thank you again.

Reply:
?? I do not get that...

Reply:
All the tests are passing in the NoJIT build, and in the JIT build, there's just a specific test with backrefs that's failing that I'm struggling to fix.